src/algebra/gradient.jl

```diff
@@ -202,3 +210,29 @@ function LazyOperators.materialize(
 ) where {O}
     return materialize(Gradient(get_args(get_args(lOp)...)), cPoint)
 end
+
+function LazyOperators.materialize(
+    lOp::Gradient{O, <:Tuple{LazyMapOver{<:CellShapeFunctions}}},
+    sidePoint::AbstractSide{Nothing, <:Tuple{FacePoint}},
+) where {O}
+    op_side = get_operator(sidePoint)
+    cellPoint = op_side(get_args(sidePoint)...)
+    return materialize(Gradient(get_args(get_args(lOp)...)), cellPoint)
+end
+
+function LazyOperators.materialize(
+    lOp::Gradient{O, <:Tuple{LazyMapOver}},
+    sidePoint::AbstractSide{Nothing, <:Tuple{FacePoint}},
+) where {O}
+    ∇x = tuplemap(x -> materialize(∇(x), sidePoint), get_args(get_args(lOp)...))
+    return MapOver(∇x)
+end
+
+function LazyOperators.materialize(
+    lOp::Gradient{O, <:Tuple{Vararg{AbstractCellFunction}}},
+    side::AbstractSide{Nothing, <:Tuple{FacePoint}},
```

Comment:
for coherence, can you name the argument `sidePoint`?

Reply:
done